libs/wire-api/src/Wire/API/Password.hs

```diff
@@ -14,38 +15,84 @@
 --
 -- You should have received a copy of the GNU Affero General Public License along
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
 
-module Wire.API.Password
-  ( Password,
-    genPassword,
-    mkSafePassword,
-    verifyPassword,
-  )
-where
+module Wire.API.Password where
```

Comment:
Haven't looked into details, but probably you want to export only the smart constructors so as to avoid opening a possibility to construct an arbitrary `Password` by hand that might be invalid.

Reply:
(maybe also rename the `Password` constructor to `UnsafeMkPassword`?)

Reply:
Good catch. I was exporting more to test that things were compatible with the previous hash function. Forgot to constrain it again 👍